### PR TITLE
[WIP] [Identity] TokenCache persistence redesign

### PR DESCRIPTION
**Pull request by schaabs**
This PR is a redesign of the mechanisms provided for persisting the token cache. 
- Removes the `EnablePersistentCache` and `AllowUnencryptedCache` options from the various `TokenCredentialOptions` classes providing them and replaces them with the `TokenCache` property.
- Adds the classes `TokenCache` and the derived class `PersistentTokenCache` to enable applications to persist the token cache either via the built in MSAL extensions mechanisms, or in a completely custom way

### CODE REVIEW

File: sdk/identity/Azure.Identity/src/MsalClientBase.cs [hunk 1]
```diff
@@ -62,54 +54,13 @@ protected async ValueTask<TClient> GetClientAsync(bool async, CancellationToken
...
-                        cacheHelper.VerifyPersistence();
-                    }
-                    else
-                    {
-                        throw;
-                    }
-                }
-
-                cacheHelper.RegisterCache(client.UserTokenCache);
+                await TokenCache.RegisterCache(async, client.UserTokenCache, cancellationToken).ConfigureAwait(false);
```
Comment (AlexanderSher) [inline sdk/identity/Azure.Identity/src/MsalClientBase.cs:59]: We probably need a try/catch here, wrapping exception with `CredentialUnavailableException`.

File: sdk/identity/Azure.Identity/src/TokenCache.cs [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+            }
+            else
+            {
+                stream.Read(data, 0, data.Length);
+            }
+
+            return new TokenCache(data);
+        }
+
+        internal virtual async Task RegisterCache(bool async, ITokenCache tokenCache, CancellationToken cancellationToken)
```
Comment (AlexanderSher) [inline sdk/identity/Azure.Identity/src/TokenCache.cs:146]: Should probably be called `RegisterCacheAsync`

**Comment by jsquire** [comment]
Hi @schaabs.  There hasn't been recent engagement on this PR.  Would you please be so kind as to let us know if this is still an active work stream or if the PR should be closed out?

**Comment by jsquire** [comment]
Since there hasn't been recent engagement, I'm going to close this out.  Please feel free to reopen if you'd like to continue working on these changes.

**Pull request closed without merging**